Src/Test.UnitTests.Sarif.PatternMatcher/SharedUtilitiesTests.cs

```diff
@@ -32,5 +34,35 @@ public void SharedUtilities_PopulateAssetFingerprint()
             fingerprint.Part.Should().Be("servers");
             fingerprint.Platform.Should().Be(nameof(AssetPlatform.SqlOnPremise));
         }
+
+        [Fact]
+        public void SharedUtilities_PopulateAssetFingerprint_WithHostList()
+        {
+            var fingerprint = new Fingerprint();
+            string[] azureHosts = new string[]
+            {
+                "resource.database.windows.net",
+                "resource.database.azure.com",
+                "mysqldb.chinacloudapi.cn",
+                "mysql.database.azure.com",
+            };
+
+            var otherAzureHosts = new List<string>
+            {
+                "mysqldb.chinacloudapi.cn",
+                "mysql.database.azure.com",
+            };
+
+            foreach (string azureHost in azureHosts)
+            {
+                SharedUtilities.PopulateAssetFingerprint(otherAzureHosts, azureHost, ref fingerprint);
+                fingerprint.Part.Should().Be("servers");
+                fingerprint.Platform.Should().Be(nameof(AssetPlatform.Azure));
+            }
+
+            SharedUtilities.PopulateAssetFingerprint(otherAzureHosts, "some-unknown-host", ref fingerprint);
+            fingerprint.Part.Should().Be("servers");
```

Comment:
>fingerprint.Part.Should(). [](http://example.com/codeflow?start=12&length=26)

this is wrong.
move this before the foreach and you will see that part will be empty.

also, if  one of the tests start  failing  you would have to debug and see which one is the wrong one.

take a look at our other tests and see  our pattern

Reply:
I made a few changes:
1. created a testCase dynamic with input and expected output
2. we compare everything and report in the end
3. we are always showing which test failed.